chain/chain/src/store.rs

```diff
@@ -1193,14 +1193,18 @@ impl<'a> ChainStoreUpdate<'a> {
         }
     }
 
+    /// Return all incoming receipts that need to be processed by shard `shard_id` at block with
+    /// block hash `block_hash`. `last_chunk_height_included` is the height of the last block that
+    /// contains a non-empty chunk for the shard before `block_hash`. The results will include
+    /// all incoming receipts sent to shard `shard_id` from the block at height
+    /// `last_chunk_height_included` + 1 to the block at `block_hash`.
     pub fn get_incoming_receipts_for_shard(
         &mut self,
         shard_id: ShardId,
         mut block_hash: CryptoHash,
```

Comment:
i'm surprised that block_hash is mutable

Reply:
yeah this piece of code is not optimal